data-products/src/firefox_newtab/sql/ads_impressions_hourly/ads_daily_engagement_by_tile_id_position/data.sql

```diff
@@ -0,0 +1,27 @@
+SELECT
+  DATE(i.submission_timestamp) AS happened_at,
+  NULL AS recommendation_id,
+  COALESCE(SAFE_CAST(i.metrics.string.ad_id AS int), -1) AS tile_id,
+  COALESCE(SAFE_CAST(i.metrics.string.ad_client_position AS int), -1) AS position,
+  SUM(CASE
+      WHEN i.metrics.string.ad_interaction = 'impression' THEN 1
+      ELSE 0
+  END
+    ) AS impression_count,
+  SUM(CASE
+      WHEN i.metrics.string.ad_interaction = 'click' THEN 1
+      ELSE 0
+  END
+    ) AS click_count,
+  0 AS save_count,
+  0 AS dismiss_count
+FROM
+  ads_backend.interaction AS i
+WHERE 
+    submission_timestamp >= '{{ batch_start }}'
+    AND submission_timestamp < '{{ batch_end }}'
```

Comment:
We'll need to do the same logic for handling live vs stable tables as exists in the other job https://github.com/Pocket/data-flows/blob/main-v2/data-products/src/firefox_newtab/sql/glean_firefox_new_tab_impressions_hourly/glean_firefox_new_tab_daily_engagement_by_tile_id_position/data.sql#L4-L24

We also need to do some filtering to SPOC desktop interactions

I think this will do it, I'll do some testing as well to get the query working

```suggestion
FROM
  {% if with_stable %}
  `moz-fx-data-shared-prod.ads_backend_stable.interaction_v1` AS i
  {% else %}
  `moz-fx-data-shared-prod.ads_backend_live.interaction_v1` AS i
  {% endif %}
WHERE 
  submission_timestamp >= '{{ batch_start }}'
  AND submission_timestamp < '{{ batch_end }}'
  AND i.metrics.string.ad_client_form_factor = 'desktop'
  AND i.metrics.string.ad_client_placement = 'newtab_spocs'
QUALIFY 
  ROW_NUMBER() OVER (
    PARTITION BY DATE(submission_timestamp), document_id
    ORDER BY submission_timestamp DESC
  ) = 1
```

I'm not sure if we can do this and the GROUP BY, we might need to put all the logic except the aggregation into a CTE and then do the GROUP BY

Reply:
Dropping in an updated suggestion, closing this one